File: Năm 2/Kì 2 đợt 2/ThuatToanATTT/Bieudienmang.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

int main()
{
    unsigned long long a,b;
    printf("Nhap a va b: ");
    scanf("%llu %llu", &a, &b);
    printf("a = %llu, b = %llu\n", a, b);
    int W = 8, t = 4, m = 31;
    unsigned long long p = 2147483647;
    int u[t], y[t];
    for (int j = 0; j < t; j++)
    {
        unsigned long long x = pow(2, W *(t - j - 1));
        u[j] = round(a / x);
        y[j] = round(b / x);
        a -= x * u[j];
        b -= x * y[j];
    }
    printf("Bieu dien qua mang a trong F_%llu voi W = %d la: (", p, W);
    for (int j = 0; j < t; j++)
    {
        if (j != t)
            printf("%d, ", u[t]);
        else
            printf("%d)\n", u[t]);
    }

    printf("Bieu dien qua mang b trong F_%llu voi W = %d la: (", p, W);
    for (int j = 0; j < t; j++)
    {
        if (j != t)
            printf("%d, ", y[t]);
        else
            printf("%d)\n", y[t]);
    }
    return 0;
}

File: Năm 2/Kì 2 đợt 2/ThuatToanATTT/.vscode/Phần 1/Bieudienmang.py
import math
p = int(input("Nhap p: "))
w = int(input("Nhap w: "))
a = int(input("Nhap a: "))
m = math.ceil(math.log(p, 2))
t = math.ceil(m/w)
array_a = []
while(t > 0):
    x = 2**(w*(t - 1))
    array_a.append(int(a/x))
    a -= int(a/x)*x
    t -= 1
print(f"Bieu dien mang cua a tren truong F_{p} voi W = {w} la: ")
print(array_a)